### ISSUE TITLE
Cursor should know how to disable ctrl-k (for terminal)

### ISSUE THREAD

**Issue by guettli** [getcursor/cursor#1756]
I use `ctrl-k` on the command line to delete all characters starting from the current position.

This is hard-wired into my brain, since I use it for 20 years.

I asked Cursor how to update the config, so that ctrl-k in the terminal does not open the smart pop-up.

It failed badly. And I found no solution, too.

**Comment by spy16x** [comment]
I had this issue too. You can goto Keybindings and search for `ctrl + k` and remove the startComposer binding there.

**Comment by Kain-90** [comment]
Same issue to me. I tried to remove all conditions that conflict with 'command+k', but the `clear` command still doesn't work in the terminal.

So far, I have to type `cl` to clear my terminal after done this `alias cl=clear`

<img width="1548" alt="image" src="https://github.com/user-attachments/assets/62794dbd-d291-420c-9580-ab16b68e096b">

**Comment by spy16x** [comment]
that keybiding is still there i think? you can right click on it and select "Remove keybinding" option.

**Comment by Kain-90** [comment]
> that keybniding is still there i think? you can right click on it and select "Remove keybinding" option.

Yes, remove is ok with it, but I don't know why setting 'when' to empty is not work.

**Comment by spy16x** [comment]
`when` only limits the scope of when this binding should work. without it, it is applicable everywhere i think.

**Comment by n13** [comment]
> Same issue to me. I tried to remove all conditions that conflict with 'command+k', but the `clear` command still doesn't work in the terminal.

Delete "Generate in Terminal" . Then it starts working.

**Comment by n13** [comment]
This is an absurly annoying bug - what would drive you to pick basically the most used terminal shortcut and override it with an AI feature??

I finally took the time today to go to settings, search for "clear terminal", click on "same shortcut"

You could also search shortcuts with the phrase "Command+K"

Then delete Generate in Terminal. 

then it started working. 

I literally don't use any other shortcuts in terminal. CMD-K is very important so we can delete all the old output in one go, and have a new clean terminal and then run whatever we're running again, only seeing the latest outputs. 

Please change this to pretty much anythiung else. Just not CMD=K in terminal. Thanks.

**Comment by eharris128** [comment]
Ran into the same issue except ctrl + shift + k. (line delete)

**Comment by stephen-last** [comment]
Find `cursorai.action.generateInTerminal` in the keyboard shortcuts and remove it.

Or, add a new key binding for it, I used `cmd + k cmd + k` (hold down `cmd` and press `k` twice).

This way you can still get the AI feature quickly, but it doesn't mess with my muscle memory for `cmd + k` to clear.

**Comment by n13** [comment]
Thanks that’s what I did. Just took me a while to find the right shortcut to disable. On 23 Nov 2024, at 01:02, Stephen Last ***@***.***> wrote:﻿
Find cursorai.action.generateInTerminal in the keyboard shortcuts and remove it.
Or, add a new key binding for it, I used cmd + k cmd + k (hold down cmd and press k twice).
This way you can still get the AI feature quickly, but it doesn't mess with my muscle memory for cmd + k to clear.

**Comment by good-epic** [comment]
I'm in the same boat but when I delete or remap or change the when condition for the ctrl+k command it inserts it into the bottom of my keybindings.json file. If I delete that and save the file. It undoes the global changes I made in the default settings. What the fuck!!!???

**Comment by jillesca** [comment]
Yep, quite annoying this mapping. I ended using @stephen-last suggestion.

**Comment by brianwarsing** [comment]
##### Open your user-level keybindings.json

<img width="612" alt="Image" src="https://github.com/user-attachments/assets/7392d2f0-5443-4c0c-aaf4-655ee14a6c6b" />

##### Add this to the very bottom

<img width="775" alt="Image" src="https://github.com/user-attachments/assets/a3bc3692-a06f-4113-add4-bba7c9c55510" />

##### Code

- NOTE: the `-` at the head of the `cursorai.action.generateInTerminal` command negates the binding
```json
 {
    "key": "cmd+k",
    "command": "-cursorai.action.generateInTerminal",
    "when": "terminalFocus && terminalHasBeenCreated || terminalFocus && terminalProcessSupported"
  }
```

Credit: https://github.com/getcursor/cursor/issues/1756#issuecomment-2494288147

Thanks @stephen-last for finding the setting, my sanity was waning

**Comment by danrice92** [comment]
Hey, can the Cursor devs maybe pick a new keybinding and quit overriding `Command+K` in general? It's a very commonly used keybinding for devs using the integrated terminal, and it broke again when I updated Cursor after I specifically defined a custom keybinding to override it.

It seems user-hostile to change my keybindings without asking me, and even more so for a shortcut that's so regularly used.

**Comment by stephen-last** [comment]
> Hey, can the Cursor devs maybe pick a new keybinding and quit overriding `Command+K` in general? It's a very commonly used keybinding for devs using the integrated terminal, and it broke again when I updated Cursor after I specifically defined a custom keybinding to override it.
> 
> It seems user-hostile to change my keybindings without asking me, and even more so for a shortcut that's so regularly used.

I second this! Exactly the same thing happened to me again a few days ago...

**Comment by reduardo7** [comment]
> ##### Open your user-level keybindings.json
> <img alt="Image" width="612" src="https://private-user-images.githubusercontent.com/150367125/430650635-7392d2f0-5443-4c0c-aaf4-655ee14a6c6b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU0Mzg0OTMsIm5iZiI6MTc0NTQzODE5MywicGF0aCI6Ii8xNTAzNjcxMjUvNDMwNjUwNjM1LTczOTJkMmYwLTU0NDMtNGMwYy1hYWY0LTY1NWVlMTRhNmM2Yi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNDIzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDQyM1QxOTU2MzNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wYThmZDQ3NzFjMTEyMGVkZTIwOTQ2Zjk0ZmUzOTIzZjc5NmVjOTdlNmI2YTM4MDY4MjEzMmYxZjY2YzcwZTE4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.yxadfzeFO9TaNvjExf8zu6j86eDkLyEZCHOkyChgrpM">
> ##### Add this to the very bottom
> <img alt="Image" width="775" src="https://private-user-images.githubusercontent.com/150367125/430650649-a3bc3692-a06f-4113-add4-bba7c9c55510.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU0Mzg0OTMsIm5iZiI6MTc0NTQzODE5MywicGF0aCI6Ii8xNTAzNjcxMjUvNDMwNjUwNjQ5LWEzYmMzNjkyLWEwNmYtNDExMy1hZGQ0LWJiYTdjOWM1NTUxMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNDIzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDQyM1QxOTU2MzNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kMWY1MTc5MDlkMDhkNjgyOTBhYzJlZjJlM2E5MzI0ZTNkZDNjM2YzNDk3NjM0NjFjZWNkYjA1NTFjNGU5YmJlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.QFANZL0uKmkamies6Fw7xGdjLGq3XCLX1CvSWmKV6dY">
> ##### Code
> * NOTE: the `-` at the head of the `cursorai.action.generateInTerminal` command negates the binding
> 
>  {
>     "key": "cmd+k",
>     "command": "-cursorai.action.generateInTerminal",
>     "when": "terminalFocus && terminalHasBeenCreated || terminalFocus && terminalProcessSupported"
>   }
> Credit: [#1756 (comment)](https://github.com/getcursor/cursor/issues/1756#issuecomment-2494288147)
> 
> Thanks [@stephen-last](https://github.com/stephen-last) for finding the setting, my sanity was waning
> 
> ##### UPDATE:
> And only two weeks later they changed it again ...
> 
> `"when": "terminalFocus && terminalHasBeenCreated || terminalFocus && terminalProcessSupported || terminalHasBeenCreated && terminalPromptBarVisible || terminalProcessSupported && terminalPromptBarVisible"`
> 
> 😭

Thank you!

My final working configuration:

```json
  {
    // https://github.com/getcursor/cursor/issues/1756#issuecomment-2781003573
    "key": "cmd+k",
    "command": "-cursorai.action.generateInTerminal",
    "when": "terminalFocus && terminalHasBeenCreated || terminalFocus && terminalProcessSupported || terminalHasBeenCreated && terminalPromptBarVisible || terminalProcessSupported && terminalPromptBarVisible"
  },
  {
    // https://github.com/getcursor/cursor/issues/1756#issuecomment-2781003573
    "key": "cmd+i",
    "command": "cursorai.action.generateInTerminal",
    "when": "terminalFocus && terminalHasBeenCreated || terminalFocus && terminalProcessSupported || terminalHasBeenCreated && terminalPromptBarVisible || terminalProcessSupported && terminalPromptBarVisible"
  },
```
- Invalidate <kbd>⌘ command</kbd> + <kbd>k</kbd> in terminal.
- Add <kbd>⌘ command</kbd> + <kbd>i</kbd> to use the `cursorai.action.generateInTerminal` command.

**Comment by nicolaipre** [comment]
Same thing goes for `Ctrl + L`. Very annoying that that it automatically jumps to the chat.

A possible solution to allow multiple uses of the hotkeys for both terminal and other Cursor functionality would be to perform actions based on the active window, e.g:

- If terminal window is active: use hotkeys for terminal actions
- If the code window is active: use hotkeys for Cursor chat etc.
